### ISSUE TITLE
User input [IN PROGRESS]

### ISSUE THREAD

**Issue by andre-d** [jtauber/dcpu16py#62]

**Comment by matze** [comment]
You can find a proposal at https://github.com/matze/dcpu16py/commit/6bf37269de99a0f78418e742edba3f8908387cc5. This works for this snippet

```
:start
    set i, 0
:readchar
    set a, [0x9000]
    ifg a, 0
        set PC, writechar
    set PC, readchar
:writechar
    set [0x9000], 0
    set [0x8000+i], a
    add i, 1
    set PC, readchar
```

But it is not very stable yet and relies on non-blocking character input (either by timeout or by starting a new thread).

**Comment by andre-d** [comment]
I have been using this one that I wrote: https://gist.github.com/2340075.  The version I have uses TerminalPlugin like yours.  After a call to updatekeys() it checks for self.term.keys (a "key stack" list).  The intention being that the terminal may write to that within the updatekeys() call...however, they could write to it at any time. Curses windows have a nodelay() function which I used (pass in 1) to ensure that the input is not blocking.  Seems to be working, I will post my branch after I cleanup and add the support into debug, pygame, and qt.
